kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -214,6 +215,24 @@ public long getLastModified() {
         throw new DatasetIOException("Cannot find last modified time of of " + dir, e);
       }
     }
+
+    // if view was marked ready more recently count it as the modified time
+    long readyTimestamp = ((FileSystemDataset<E>)dataset).getReadyTimestamp(getConstraints());
+    if(lastMod < readyTimestamp) {
+      lastMod = readyTimestamp;
+    }
+
     return lastMod;
   }
+
+  @Override
+  public void signalReady() {
+    ((FileSystemDataset<E>)dataset).signalReady(getConstraints());
+  }
+
+  @Override
+  public boolean isReady() {
+    long readyTimestamp = ((FileSystemDataset<E>)dataset).getReadyTimestamp(getConstraints());
+    return readyTimestamp != -1;
```

Comment:
It could be and would probably operate the same way, the ready timestamp is -1 (by the API in the SignalManager) when the view hasn't been signaled as ready so this feels more correct. Zero would be a valid timestamp for a ready signal exactly at the epoch, and in the same vein negative values shouldn't be possible.